src/main/scala/com/cloudant/clouseau/IndexService.scala

```diff
@@ -199,6 +206,26 @@ class IndexService(ctx: ServiceContext[IndexServiceArgs]) extends Service(ctx) w
     }
   }
 
+  private def commitPurgeSeq(newSeq: Long) {
```

Comment:
This method does not set the committing flag and so could allow concurrent attempts to commit, leading to a broken index (one with the wrong values).

This work should instead be done in the preceding `commit` function.